float64 quad_x
float64 quad_y
float64 quad_z
